elyra/pipeline/component.py

```diff
@@ -24,21 +25,11 @@
 from traitlets.config import LoggingConfigurable
 
 
-class ComponentProperty(object):
+class ComponentParameter(object):
     """
     Represents a single property for a pipeline component
     """
 
-    ref: str
-    name: str
-    type: str
-    value: str
-    description: str
-    control: str
-    control_id: str
-    items: list
-    required: bool
-
     def __init__(self, ref: str, name: str, type: str, value: str, description: str, required: bool = False,
                  control: str = "custom", control_id: str = "StringControl", items: List[str] = []):
```

Comment:
The default value for `items` needs to be updated or removed.  Having a mutable object as a default value will result in side effects.

I'll post a commit to address this.